source/GameEvent.cpp

```diff
@@ -89,18 +90,14 @@ void GameEvent::Save(DataWriter &out) const
 		conditionsToApply.Save(out);
 		
 		for(const System *system : systemsToUnvisit)
-			if(system && !system->Name().empty())
-				out.Write("unvisit", system->Name());
+			out.Write("unvisit", system->Name());
```

Comment:
I'm not sure how `Set<T>::Get` will return `nullptr` without there being worse issues (like memory allocation failures) that ES can't prevent anyway.

Reply:
This is more of a general "just in case" thing.

The nullptr-checks are cheap and and it is considered good practice in the school where I learned to program to always check pointers from outside the function for NULL before dereferencing. Even if you are 100% sure today that they are not needed.

The main argument there is not failing hardware (although it is also a good argument), but bugs from obscure corner-cases that were overlooked by the programmers working on the code (even if the corner-case itself comes from a future code-change).

Most of the time the check is indeed not needed, but the value of handling some pointer in the obscure corner-case gracefully was considered high enough to warrant all the other checks that are never needed.

Another argument is review-effort. I can lookup the systemsToUnvisit data-structure (and notice that is a vector) and then lookup all the locations where systemsToUnvisit gets filled (and then verify for all of those locations that there never will be a situation where a nullptr gets added).
But if this code just handles nullpointers gracefully then I don't need to do all that additional review-effort to do a proper review.